src/node/types.h

```diff
@@ -37,7 +37,7 @@ struct BlockCreateOptions {
      * scriptSig, witness and outputs. This must include any additional
      * weight needed for larger CompactSize encoded lengths.
      */
-    size_t coinbase_max_additional_weight{4000};
+    size_t coinbase_max_additional_weight{0};
```

Comment:
I don't think setting `coinbase_max_additional_weight` to `0` is the right approach. 

I'll comment on #21950 first.

Reply:
Switched to the approach you suggested.